### PR TITLE
Implemented deprecated commands

### PR DESCRIPTION
**Pull request by Valansch**

### CODE REVIEW

**Review by mheguy — DISMISSED** [review]
In testing as a player, this logged the message rather than printing.

**Comment by mheguy** [comment]
Stylistically, we should probably stick to only single newlines to break up code blocks.

**Comment by Valansch** [comment]
@plague006 
>In testing as a player, this logged the message rather than printing.
no? As a player this will print to chat. As server this will print to console and log it.

**Comment by grilledham** [comment]
`game.player` is only accessible inside a command. You are using it inside an event.

**Comment by Valansch** [comment]
My bad, you are right. I thought of on_console_command as a command 🙄

File: features/custom_commands.lua [hunk 1]
```diff
@@ -11,7 +11,21 @@ local Command = require 'utils.command'
...
+local deprecated_command_alternatives = {
+    ['silent-command'] = 'sc'
+}
+
+Event.add(defines.events.on_console_command, function (event)
+        local alternative = deprecated_command_alternatives[event.command]
+        if alternative then
+            local print = log
+            if event.player_index then
+                print = game.players[event.player_index].print
```
Comment (grilledham) [inline features/custom_commands.lua]: We normally use `Game.get_player_by_index` when getting player by index

Comment (Valansch) [reply features/custom_commands.lua]: Fixed

File: features/custom_commands.lua [hunk 2]
```diff
@@ -11,7 +11,21 @@ local Command = require 'utils.command'
...
+}
+
+Event.add(defines.events.on_console_command, function (event)
+        local alternative = deprecated_command_alternatives[event.command]
+        if alternative then
+            local print = log
+            if event.player_index then
+                print = Game.get_player_by_index(event.player_index).print
+            end
+            print(string.format('Warning! Usage of the command /"%s" is deprecated. Please use "%s" instead.', event.command, alternative))
```
Comment (mheguy) [inline features/custom_commands.lua]: `command /"%s" is` should probably be `command "/%s" is`, it looks weird to have the slash outside of the quotes. `use "%s" instead` should probably be `use "/%s" instead` to match the style of the previous.

**Review by mheguy — APPROVED** [review]
Looks good to me.

**Pull request merged**